### ISSUE TITLE
Service Fabric: Client disconnects on WS/SSE

### ISSUE THREAD

**Issue by darrenfurr** [aspnet/SignalR#2613]
Server:
Microsoft.AspNetCore - 2.1.1
Microsoft.AspNetCore.SignalR - 1.0.1
Microsoft.ServiceFabric - 6.2.301
Microsoft.ServiceFabric.AspNetCore.Kestrel - 3.1.301

Client:
"@aspnet/signalr": "^1.0.0"

Operating Systems:
Windows 10 Enterprise Version 1803

Browsers:
Chrome 67.0.3396.99 (Official Build) (64-bit)

When utilizing SignalR with the latest version of ServiceFabric - our angular client disconnects after 30 seconds.

Error on the client:
`Error: Connection disconnected with error 'Error: Server timeout elapsed without receiving a message from the server.'.`

Attempted troubleshooting steps:

Added KeepAliveInternal & set to 5 seconds - still disconnects

- Switched to SSE transport - still disconnects
- Deployed a separate Kestrel server - works fine.
- Switched to LongPolling for SignalR - works fine.

Any help here would be appreciated.

**Comment by BrennanConroy** [comment]
Could you collect server logs? https://github.com/aspnet/SignalR/wiki/Diagnostics-Guide#server-side-logging

Does Service Fabric timeout long running connections possibly?

**Comment by darrenfurr** [comment]
@BrennanConroy - I'm working on spinning up a PoC, gathering logs, & will send a link to the repo.

**Comment by darrenfurr** [comment]
@BrennanConroy - it appears that after spinning up a PoC, this issue was caused by my autofac implementation. It appears to be related to how I was registering the autofac container with aspnetcore.

Inside the Configure method, I was doing:
`app.ApplicationServices = new AutofacServiceProvider(this.Container);`

And then inside of ConfigureServices:
`return services.BuildServiceProvider();`

I removed the app.ApplicationServices setter & inside ConfigureServices did:
`return new AutofacServiceProvider(this.Container);`

And - it's all working now.
